File: SYP.Web/Modules/Order/TieredDiscountSettings/Editor/TieredDiscountSettingsEditorForm.cs
namespace SYP.Order.Forms;

[FormScript("Order.TieredDiscountSettingsEditor")]
[BasedOnRow(typeof(TieredDiscountSettingsRow), CheckNames = true)]
public class TieredDiscountSettingsEditorForm
{
    public decimal MinAmount { get; set; }
    public decimal DiscountPercentage { get; set; }
    public bool IsActive { get; set; }
    public int DisplayOrder { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Order/TieredDiscountSettingsEditorForm.ts
import { BooleanEditor, DateEditor, DecimalEditor, initFormType, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";

export interface TieredDiscountSettingsEditorForm {
    MinAmount: DecimalEditor;
    DiscountPercentage: DecimalEditor;
    IsActive: BooleanEditor;
    DisplayOrder: IntegerEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
}

export class TieredDiscountSettingsEditorForm extends PrefixedContext {
    static readonly formKey = 'Order.TieredDiscountSettingsEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TieredDiscountSettingsEditorForm.init) {
            TieredDiscountSettingsEditorForm.init = true;

            var w0 = DecimalEditor;
            var w1 = BooleanEditor;
            var w2 = IntegerEditor;
            var w3 = DateEditor;

            initFormType(TieredDiscountSettingsEditorForm, [
                'MinAmount', w0,
                'DiscountPercentage', w0,
                'IsActive', w1,
                'DisplayOrder', w2,
                'InsertDate', w3,
                'InsertUserId', w2,
                'UpdateDate', w3,
                'UpdateUserId', w2
            ]);
        }
    }
}